### PR TITLE
Added cmake find_package option

### PR DESCRIPTION
**Pull request by bmarques1995**
This commit has added the option to find package to D3D12MA, where you can call `find_package(GPUOpen COMPONENTS D3D12MemoryAllocator)`

### CODE REVIEW

File: .gitignore
Comment (adam-sawicki-a) [inline .gitignore:1]: Changes in this file seem unrelated to the pull request.

File: README.md
Comment (adam-sawicki-a) [inline README.md:1]: Changes in this file seem unrelated to the pull request.

File: src/cmake/config.cmake.in [hunk 1]
```diff
@@ -0,0 +1,13 @@
+include(CMakeFindDependencyMacro)
+
+find_library(D3D12_LIB d3d12.lib REQUIRED)
```
Comment (adam-sawicki-a) [inline src/cmake/config.cmake.in:3]: What if someone wants to load D3D12 dynamically using `LoadLibrary`, `GetProcAddress` rather than statically linking to these libraries? I wouldn't require them as a dependency, rather let the user link/load them themselves.

**Review by adam-sawicki-a — CHANGES_REQUESTED** [review]
What is the overall purpose of this new mechanics? Can you please show some example of how using D3D12MA through Cmake would look like after this change and what is the difference compared to what we have now?

File: src/cmake/config.cmake.in (hunk 1 above)
Comment (bmarques1995) [reply src/cmake/config.cmake.in:3]: Hi, @adam-sawicki-a, let me explain your demands, the unrelated files is easy, i can move them to next PRs, the config.cmake.in, if you wish, is okay to remove the dependencies, for the final result, in addition of the result files in the installation, because this PR is only for the cmake installation process, in the install folder will be created another folder, in share, with the D3D12MA config and target files, (each `-` indicates one level dir):
`
- share
-- cmake
--- D3D12MemoryAllocator
---- D3D12MemoryAllocatorConfig.cmake
---- D3D12MemoryAllocatorConfigVersion.cmake
---- D3D12MemoryAllocatorTargets.cmake
---- D3D12MemoryAllocatorTargets-debug.cmake
`

With these files, is possible, in cmake to use the command: 
`
find_package(D3D12MemoryAllocator)
#linkage method
target_link_libraries(${TARGET_NAME} PUBLIC GPUOpen::D3D12MemoryAllocator)
`

This principle is used in VMA, but is a little different because the lib is header only.

File: README.md
Comment (bmarques1995) [reply README.md:1]: By the way, this one could u do this gentleness, it's advertising to this repository, and inspired by it i came here to do this PR, so, could u add this content to the official readme? I will add the file to be only downloaded
[README.md](https://github.com/user-attachments/files/15837361/README.md)

**Comment by bmarques1995** [comment]
> What is the overall purpose of this new mechanics? Can you please show some example of how using D3D12MA through Cmake would look like after this change and what is the difference compared to what we have now?

Are these all your change requests?

**Comment by bmarques1995** [comment]
> > What is the overall purpose of this new mechanics? Can you please show some example of how using D3D12MA through Cmake would look like after this change and what is the difference compared to what we have now?
> 
> Are these all your change requests?

For this Pull Request, yes

File: CMakeLists.txt
Comment (bmarques1995) [inline CMakeLists.txt:1]: @adam-sawicki-a this version is only an example of structure, that u can add version to cmake, if u wish i can write another version, but have the version set is required to the cmake package

**Comment by adam-sawicki-a** [comment]
Thank you for this contribution!
